processor.go

```diff
@@ -223,13 +263,18 @@ func (pro *Processor) applyCandidate(proposal *message.Proposal) error {
 	// extension is to the external module, which can do additional checks such
 	// as validating the payload
 	err = pro.graph.Extend(proposal.Candidate)
+	// previous checks are successful to this point -> rest - including payload validation
+	// are handled by the external component implementing Graph interface
+	// explained also by the comment
 	if err != nil {
 		return rich.Errorf("could not extend graph: %w", err)
 	}
 
 	// 7) check if this particular proposal has already been cached, or if
 	// there is a double proposal situation being created
 	err = pro.cache.Proposal(proposal)
+	// not sure what does the Proposal method for Cache interface should do in this case
+	// but it should check if there is a double record in the cache record and add it if not present
```

Comment:
If the proposal has already been cached, this is just a retransmission, and we might want to ignore it. If there is an alternative valid proposal, we should create a slashing challenge, as the validator / leader made two proposals, which is illegal.